### ISSUE TITLE
Clipboard problem

### ISSUE THREAD

**Issue by embryo10** [kivy/kivy#3954]
This is a copy (with some additions) of a [post at Google Groups](https://groups.google.com/d/msg/kivy-users/pnXpQNwhjYc/2ZMjxl8vAgAJ).
 Python 2.7.10 / kivy 1.9.0
Windows XP SP3

Using Windows I was trying `Clipboard.get_types()` to see if the clipboard got text.
That was returning a list with `"T", "e", "x", "t", "/" `...
So, I used an `if ''.join(Clipboard.get_types()) == 'text/plain'` to check and it was working OK.
With the script that is.

Compiling the script with PyInstaller 2.1 (v3.0 does not work with 1.9.0) produced an executable that didn't work (not all the executable, just the clipboard part of it)
After some tries, I found out that the `Clipboard.get_types()` was now returning `![u'text/plain;charset=utf-8', u'BlaBla Editor Data']` (the BlaBla is the editor that made the text copy)
I solve the problem by making the if check bigger by adding a regular expression:
```
if (''.join(Clipboard.get_types()) == 'text/plain' or
        re.search(r".+text/plain.+", str(Clipboard.get_types()), re.DOTALL)):
```
but I was wondering if there was a simpler and more elegant way to check for text...

Why it behaves differently from the script and from the compiled script?

**Comment by dessant** [comment]
@embryo10, Clipboard.get_types() should return the correct value in master, please reopen if you still have the issue after updating.

**Comment by embryo10** [comment]
Tried to update kivy to 1.9.1 twice but didn't succeed.
I had the portable windows version 1.8.0 (with python 2.7.3 I think) that I managed to update to 1.9.0 (python 2.7.10).
After that If I was updating kivy, no project was working.
I have Windows Xp SP3 and use Intellij IDEA as an IDE.
I also tried to update from inside the IDEA, but with no luck.
So, I don't think I will reopen this ...

**Comment by dessant** [comment]
Actually i'm only aware of the first issue being fixed (was returning a list with "T", "e", "x", "t", "/"), but i'm not familiar with the second one.

**Comment by dessant** [comment]
@embryo10 does the returned value change to `[u'text/plain;charset=utf-8', u'BlaBla Editor Data']` only after packaging?

**Comment by embryo10** [comment]
@dessant Yes, that was what the executable was getting with `Clipboard.get_types()` and it was `![u'text/plain; charset=utf-8',u'BlaBla Editor Data']`

**Comment by embryo10** [comment]
Strangely  enough, the `Clipboard.get_types()[0]` was returning `u'BlaBla Editor Data'

**Comment by dessant** [comment]
It would help if you could confirm the exact provider being used after packaging, by default kivy selects winctypes on windows. This info should appear in the logs while copy/pasting.

**Comment by akshayaurora** [comment]
I think there is enough of a demand for the portable package that we should
bring it back, @embryo10 could you open a separate issue for that ?

**Comment by dessant** [comment]
@akshayaurora, we've discussed having a portable package as part of a kivy designer distribution, centered around it, the issue should probably be tracked there.
@embryo10 btw, have you tried following the updated intallation guide and installing the wheels? Not in the portable package, but system python. There are also nightly wheels, which should enable you to test master.

**Comment by embryo10** [comment]
@dessant 
- How can I get logs while copy/pasting after packaging?
- I will try again with the installations (when I get some time to securely backup the system) and report back.
Looking again at the [downloads](https://kivy.org/#download), I see that XP are not mentioned.

**Comment by embryo10** [comment]
Tried the installation again.
Removed all kivy stuff (except python) and `python -m pip install kivy`
It said it was successful but didn't update to 1.9.1
The dependencies was OK.
Kept working
Tried `python -m pip install --upgrade kivy`
It updated to 1.9.1 but stopped working.
Back to 1.9.0.

**Comment by dessant** [comment]
Ok, so maybe there are issues when installing the wheels in the old portable package.

Let's move forward and test the things we've talked about with 1.9.0. Please cheach the logs as you copy/paste and share which provider is being used before and after packaging with pyinstaller.

**Comment by dessant** [comment]
@embryo10, make sure to have `EXE(console=True...` in your spec, then just run the exe from your console, and the logs should be visible.

**Comment by embryo10** [comment]
As script: `"Provider: winctypes"`
Packaged: `"Provider: pygame(['clipboard_winctypes'] ignored"`
There is also a `"WARNING: library libmtdev.so.1 required via ctypes not found"` message while compiling..
Where should I get this library?
Is it supposed to be in the system already?

**Comment by dessant** [comment]
@embryo10, ok, the issue is that pygame has been deprecated, and the 1.9.0 portable package contains some new dependencies (sdl2), pygame is not shipped at all.
You have updated kivy from the 1.8.0 portable package so your environment is messed up and you're also using an old pyinstaller version, so yeah, too many variables. :)

Your next step could be to download a fresh copy of the 1.9.0 portable package, and install the latest pyinstaller in it.
https://kivy.org/downloads/1.9.0/Kivy-1.9.0-py2.7-win32-x86.exe

Starting from this package, updating to kivy master should also work, you just have to follow these instructions:
https://github.com/kivy/kivy/blob/stable-1.9.0/doc/sources/installation/installation-windows.rst#use-development-kivy

**Comment by embryo10** [comment]
Thank you for your help, but is there an kivy package without python in it?
I already have many libraries installed and I would like to avoid re-installing...

**Comment by dessant** [comment]
I don't think it's worth spending time on it, there are many differences between the 1.8.0 and 1.9.0 package, just install your deps again with pip.

**Comment by dessant** [comment]
And if you have to do it anyway, you might as well not use the portable package, and install python properly, then kivy installation is a piece of cake:
https://www.python.org/ftp/python/2.7.11/python-2.7.11.msi
https://kivy.org/docs/installation/installation-windows.html#installation

Make sure you select to add python to PATH during installation time.

**Comment by embryo10** [comment]
Thanks.
I'll try it, once I find the time.
